### PR TITLE
Remove filter operation for superlu-dist to avoid permission issues on read-only systems

### PR DESCRIPTION
**Pull request by liuyangzhuan**

### CODE REVIEW

**Review by wspear — DISMISSED** [review]
This worked in my local test.

**Comment by alalazo** [comment]
Pinged @sshudler since he added the logic, and @tldahlgren if she wants to chime in and comment why setting RPATHs as done before might be needed during tests.

**Comment by alalazo** [comment]
@spackbot run pipeline

File: var/spack/repos/builtin/packages/superlu-dist/package.py
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/superlu-dist/package.py]:
```suggestion
        superludriver = join_path(self.prefix.lib, 'EXAMPLE', 'pddrive')
```

**Review by tldahlgren — COMMENTED** [review]
Suggest using join for the full path.

**Review by tldahlgren — APPROVED** [review]

File: var/spack/repos/builtin/packages/superlu-dist/package.py
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/superlu-dist/package.py:141]: Is `pddrive` automatically built and copied to that subdirectory of install `lib` as part of the build process?   If so, the method should check for the version and output a `skipping` message that there are no tests for versions before it.  The `cache_test_sources` method should also be removed.

If `pddrive` is not automatically built, then removing the above code removes the process that builds it.

**Review by tldahlgren — CHANGES_REQUESTED** [review]
On which version(s) has this been tested `spack test run superlu-dist`?

File: var/spack/repos/builtin/packages/superlu-dist/package.py
Comment (liuyangzhuan) [reply var/spack/repos/builtin/packages/superlu-dist/package.py:141]: That's a good point. Let me add the version checking, as pddrive is automatically built after a certain version.

Comment (liuyangzhuan) [reply var/spack/repos/builtin/packages/superlu-dist/package.py:141]: @tldahlgren Can you take a look at this again? I removed unused lines and added version detection in the test.

**Review by tldahlgren — COMMENTED** [review]
That's a new way of flagging the test as being skipped.  You could simply check the version at the top of `test` and report `Skipping test: only run for v7.2.0 on` (or something along those lines).  Doing so would make it easier to identify the package as one that needs to be updated if/when some other changes are made for better reporting of skipped tests.

My reservation regarding this approach is how the results will appear in the outputs for each `run_test` will be reported to CDash (once the relevant PR is merged).  An example for `superlu` is at https://cdash.spack.io/viewTest.php?onlypassed&buildid=1202791.  

I can merge this PR as-is if you prefer.  Just let me know.

**Comment by liuyangzhuan** [comment]
Hi, @tldahlgren. Thanks for the suggestion, which makes sense to me. However, my bandwidth on working on spack stuff is quite limited and I hope you could accept the PR as is.

**Review by tldahlgren — APPROVED** [review]

**Pull request merged**
